### PR TITLE
Replace scrollback implementation with scroll anchoring outside of `full` windowing mode

### PR DESCRIPTION
**Pull request by krassowski**
## References

- Includes https://github.com/jupyterlab/jupyterlab/pull/18504
- Fixes https://github.com/jupyterlab/jupyterlab/issues/18470
- Closes https://github.com/jupyterlab/jupyterlab/pull/18511
- Closes https://github.com/jupyterlab/jupyterlab/issues/18468

Quotes below are @jasongrout's description copied from #18504

> This is a follow up to https://github.com/jupyterlab/jupyterlab/pull/18503, which this PR includes. After #18503, the viewport does not shift based on changes happening outside the viewport, but the active cell can still shift when content inside the viewport changes. This PR resolves this issue.
>
> Addresses https://github.com/jupyterlab/jupyterlab/issues/18468#issuecomment-3893440330

## Code changes

- drop the custom scrollback implementation for windowing modes other than `full` in favor of native scroll anchoring with better behaviour implemented as:
    > Make active cells (or descendants thereof) the only elements able to be scroll anchors in the notebook. This means that the browser will not anchor the scroll to an element inside an inactive cell.
- increase the timeout used to deactivate the scrollback (acting the same as anchor but in `full` windowing mode) to 3 seconds without motion; we could probably make it infinite but I am not confident enough with the implementation to do so yet; 3 seconds appears to be a reasonable threshold
- resolve the promise returned by `scrollToItem` method as soon as scrolling end, not when the timeout used for scrollback method of `full` windowing mode expires
- separate the test for anchoring from the test of scrolling cell we advanced to into view; previously a single test was testing both behaviours in a way which was hard to wrap one's head around
- improve the reliability of the `positionCellPartiallyBelowViewport` utility which lead to flaky test due to imprecision of mouse wheel operations

## User-facing changes

The active cell will be kept in view more reliably, using browser-platform native solution rather than workarounds, fixing some edge cases with scrolling.

> In Chrome-based browsers or Firefox, the active cell should not be shifted up or down by other cells changing their size or gaining new output. [...]
>
> Before:
>
> https://github.com/user-attachments/assets/67c94546-b131-498c-8b0d-891bfd81cdb8
>
> After:
> 
> https://github.com/user-attachments/assets/297e2dff-8190-4c77-8751-869fac6b9892
>
> Note that the browser decides that when you are scrolled all the way to the very top, you should stay at the top. The moment you scroll just a bit down, the active cell is then anchored in the viewport.

Safari users won't see the benefit until they upgrade to a future version which enables support for `overflow-anchor`. This was recently enabled in Safari Technology Preview v238:
- https://github.com/WebKit/WebKit/pull/58575
- https://developer.apple.com/documentation/safari-technology-preview-release-notes/stp-release-238
- https://caniuse.com/?search=overflow-anchor
- The earliest non-TP Safari version that could support `overflow-anchor` would likely be 26.5, TBC

## Backwards-incompatible changes

None

## AI usage

No

### CODE REVIEW

**Comment by krassowski** [comment]
@jasongrout do you have any thoughts on this?

**Review by jasongrout — APPROVED** [review]
I think your extra commit looks great! I just tried it out manually as well.

(I think I kept not looking at this PR because it was draft.)

**Comment by krassowski** [comment]
I split the test as previously proposed in https://github.com/jupyterlab/jupyterlab/pull/18504#issuecomment-3905243623.

The anchoring test variant for windowed notebook mode fails because scroll anchoring is disabled for `full` windowing mode. It passes for `contentVisibility` (default):

[non-windowed.webm](https://github.com/user-attachments/assets/3c7ac355-03eb-4fe2-8735-0760752790ab)

I think that we should also explore removing the `overflow-anchor: none` application in the windowed mode. Maybe it is not needed now that we use browser-native scroll anchoring.

**Comment by krassowski** [comment]
> I think that we should also explore removing the `overflow-anchor: none` application in the windowed mode

In a quick test, this did not suffice. Maybe we do need to keep the old scrollback logic but only use if for `full` windowed mode as a legacy solution.

File: galata/test/jupyterlab/utils.ts [hunk 1]
```diff
@@ -39,12 +39,53 @@ export async function positionCellPartiallyBelowViewport(
...
+  // Initial scroll estimate
+  const notebookBbox = await scroller.boundingBox();
   await page.mouse.move(notebookBbox.x, notebookBbox.y);
-  const scrollOffset =
-    cellBbox.y - notebookBbox.y - notebookBbox.height + cellBbox.height * ratio;
-  await page.mouse.wheel(0, scrollOffset);
+
+  const initialScrollOffset =
+    cellBbox.y - notebookBbox.y - notebookBbox.height + targetPixels;
+  await page.mouse.wheel(0, initialScrollOffset);
```
Comment (krassowski) [inline galata/test/jupyterlab/utils.ts:71]: This was the cause of flakiness previously as the positioning with mouse wheel was not always accurate; now using a few iterations to ensure pixel-perfect positioning the tests are reliably passing.

**Comment by krassowski** [comment]
The failing galata test is related to the changes. It is unlikely to cause issues on runtime, but it would potentially lead to a change for galata users. This is because galata's `runActiveNotebookCellByCell` awaits for scrolling to complete and `scrollToItem` does not resolve its promise until the timeout (which I propose to increase) lapses.

I think we should decouple `scrollToItem` promise from the scrollback timeout as these two distinct things.
